### PR TITLE
fix(typedefs): Add type for autoEnd on LogConfig.

### LINKED ISSUE #9590: 'autoEnd' does not exist in type 'Partial<LogConfig>'
Doing this:
```js
// @ts-check
function someFn() {
  const log = Cypress.log({ autoEnd: false })
}
```
...results in this:
```js
Argument of type '{ displayName: string; message: string; autoEnd: boolean; }' is not assignable to parameter of type 'Partial<LogConfig>'.
  Object literal may only specify known properties, and 'autoEnd' does not exist in type 'Partial<LogConfig>'.
```

### PR DESCRIPTION
**Pull request by linkel**
- Closes #9590

### User facing changelog

Add type for autoEnd on LogConfig.

### Additional details

I wanted to make a small contribution to Cypress and get the dev environment set up. The only change I made was adding the type.

#### Side notes about Cypress development:
After I made the change, I built the package and built the binary, and in my test project I pointed to the package in package.json and had to copy paste the binary into my cypress cache in order to successfully `yarn` on my test project, otherwise it would try to download the binary online and fail since the version didn't exist. 

I also had to downgrade my node version to avoid a build error in node-sass. 

### How has the user experience changed?
Now it doesn't produce a type error and shows autoEnd correctly typed. 

image

### PR Tasks

- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by linkel** [comment]
Are the three circleci failures expected?

**Comment by jennifer-shehane** [comment]
@linkel Yah those are some flaky tests.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
It would be nice to have a comment here explaining what autoEnd is for, like the other props have. 

I think this is only used to set autoEnd: false for `.pause()` command correct? The idea is that the command doesn't 'end' exactly. 

Maybe understanding your use case for why you need to use this prop would be helpful.

**Comment by jennifer-shehane** [comment]
@linkel Do you still have time to address the changes requested in this PR? We will have to close this PR due to inactivity otherwise.

File: cli/types/cypress.d.ts [hunk 1]
```diff
@@ -5233,6 +5233,7 @@ declare namespace Cypress {
   interface LogConfig extends Timeoutable {
     /** The JQuery element for the command. This will highlight the command in the main window when debugging */
     $el: JQuery
+    autoEnd: boolean
```
Comment (flotwig) [inline cli/types/cypress.d.ts:5236]: Can you add a typedoc for this?

**Review by flotwig — CHANGES_REQUESTED** [review]

**Comment by emilgoldsmith** [comment]
#15076 was now merged so this should be able to be closed

**Comment by bahmutov** [comment]
closing as duplicate

**Pull request closed without merging**